### ISSUE TITLE
PHP Fatal error:  Call to a member function getDBKey() on null

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticMediaWiki#2077]
### Setup and configuration

- MediaWiki 	1.23.15
- PHP 	5.6.26-0+deb8u1 (apache2handler)
- MySQL 	5.5.52-0+deb8u1
- Semantic MediaWiki	2.4.3 (6661acd) 00:37, 30. Nov. 2016

### Issue
When running `php rebuildData.php -v --report-runtime --with-maintenance-log --ignore-exceptions --exception-log="/var/log/mediawiki/"` it always stopped at the respective object ID with
```
PHP Fatal error:  Call to a member function getDBKey() on null in /.../extensions/SemanticMediaWiki/src/SQLStore/QueryDependency/EntityIdListRelevanceDetectionFilter.php on line 145
```
This was caused by a deleted file. (Datei:Camera.png). After I disposed the object ID the script was able to continue and finish its job.

To get a backtrace was not possible. Though I know that the object ID disposal was introduced for this very issue I report this. Perhaps something may be done to avoid such a fatal in the first place.

**Comment by mwjames** [comment]
>I know that the object ID disposal was introduced for this very issue

:+1: 

> Semantic MediaWiki 2.4.3 (6661acd) 00:37, 30. Nov. 2016
>PHP Fatal error:  Call to a member function getDBKey() on null 
> Perhaps something may be done to avoid such a fatal in the first place.

Has been fixed in 2.5 with https://github.com/SemanticMediaWiki/SemanticMediaWiki/commit/22bc7c6c55c7a6c0312a345adfa7c11cf1358558#diff-4a266932f9a396d7bb2997cb82b5d020R145.

**Comment by kghbln** [comment]
> Has been fixed in 2.5 with 22bc7c6#diff-4a266932f9a396d7bb2997cb82b5d020R145.

Great, these are the best bugs. :)
